pkg/ctl/cmdutils/configfile.go

```diff
@@ -180,12 +180,12 @@ func NewCreateClusterLoader(provider *api.ProviderConfig, spec *api.ClusterConfi
 		}
 
 		return ngFilter.CheckEachNodeGroup(l.spec.NodeGroups, func(i int, ng *api.NodeGroup) error {
-			if ng.AllowSSH && ng.SSHPublicKeyPath == "" {
+			if ng.SSH.Allow && ng.SSH.PublicKeyPath == nil {
 				return fmt.Errorf("--ssh-public-key must be non-empty string")
 			}
 
 			if cmd.Flag("ssh-public-key").Changed {
-				ng.AllowSSH = true
+				ng.SSH.Allow = true
```

Comment:
This should be sufficient:
```Go
				if rng.SSH.PublicKeyPath == "" {
								return fmt.Errorf("--ssh-public-key must be non-empty string")
				}
```

Reply:
Ah, I thought that if one enables allow-ssh then the keypath would be ~/.ssh/id_rsa.pub by default. Should we then always force the user to define which key?

Reply:
> I thought that if one enables allow-ssh then the keypath would be ~/.ssh/id_rsa.pub by default.

Yes, indeed. What I'm saying is that we should move that empty string check here.